### DISCUSSION TITLE
Does SAM support a blue/green deployment pattern for ECS services running on Fargate?

### DISCUSSION THREAD

**Discussion by kduvzc** [aws/serverless-application-model#3521] [Questions] [answered]
According to [Deploying serverless applications gradually](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/automating-updates-to-serverless-apps.html) found in the [AWS Serverless Application Model Developer Guide](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html), SAM can be used to achieve a blue/green deployment pattern for Lambda functions by leveraging Codedeploy behind the scenes.

I'm kinda struggling to understand whether the same can be done for ECS services running on Fargate. Is it be possible to use SAM to deploy an ECS service via Codedeploy in a blue/green manner? I suppose the final outcome should allow us to implement what is described in [Blue/Green deployment with CodeDeploy](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-type-bluegreen.html) taken from AWS ECS official docs.

**Comment by jfuss** [comment, marked as answer]
SAM doesn't have any resources for Fargate, therefore not something we support.

**Reply by kduvzc** [reply]
Thanks for clarifying. This wasn't really straight-forward as AWS clearly states [here](https://aws.amazon.com/serverless/#:~:text=Serverless%20services%20on%20AWS) that AWS Fargate is one their core serverless technologies. Maybe support might pop-up at a later stage in the roadmap?
